internal/pkg/db/memory/addressables_test.go

```diff
@@ -0,0 +1,737 @@
+	User:       "Yoshi",
+	Password:   "MarioIsGettingHeavy",
+	Topic:      "dinosaur-things",
+}
+
+// addressable2 is a pre-constructed addressable with non-unique properties. The properties which can lead to multiple
+// results when invoking the client, such as GetAddressablesByPort, have matching properties with addressable3. This
+// helps with testing situations where multiple addressables are returned
+var addressable2 = contract.Addressable{
+	Timestamps: contract.Timestamps{
+		Created:  789,
+		Modified: 456,
+		Origin:   123,
+	},
+	// This is a valid UUID which allows us to ensure that the ID used is not replaced.
+	Id:         "1d874125-df04-4718-bd81-b550019b74aa",
+	Name:       "SecondAddressable",
+	Protocol:   "http",
+	HTTPMethod: "GET",
+	Address:    "127.0.0.1",
+	Port:       8080,
+	Path:       "/home",
+	Publisher:  "Nintendo",
+	User:       "Link the hero time",
+	Password:   "RescueZelda",
+	Topic:      "hyrule-events",
+}
+
+// addressable3 is a pre-constructed addressable with non-unique properties. The properties which can lead to multiple
+// results when invoking the client, such as GetAddressablesByPort, have matching properties with addressable2. This
+// helps with testing situations where multiple addressables are returned
+var addressable3 = contract.Addressable{
+	Timestamps: contract.Timestamps{
+		Created:  789,
+		Modified: 456,
+		Origin:   123,
+	},
+	// This is a valid UUID which allows us to ensure that the ID used is not replaced.
+	Id:         "cb7d56ca-527d-448b-91f0-bc27f9891b90",
+	Name:       "ThirdAddressable",
+	Protocol:   "stomp",
+	HTTPMethod: "PUT",
+	Address:    "127.0.0.1",
+	Port:       8080,
+	Path:       "/castle",
+	Publisher:  "Nintendo",
+	User:       "Gannon",
+	Password:   "GetTriforce",
+	Topic:      "hyrule-events",
+}
+
+func TestClient_AddAddressableUUID(t *testing.T) {
+	client := NewClient()
+	generatedId, err := client.AddAddressable(contract.Addressable{})
+	if err != nil {
+		t.Errorf("Unexpected error: %v", err)
+		return
+	}
+
+	// Verify the ID is not blank and is a valid UUID
+	_, err = uuid.Parse(generatedId)
+	if err != nil {
+		t.Errorf("The generated UUID is not valid: %s", generatedId)
+	}
+}
+
+func TestClient_GetAddressables(t *testing.T) {
+
+	tests := []struct {
+		name                     string
+		prePopulatedAddressables []contract.Addressable
+		want                     []contract.Addressable
+		wantErr                  bool
+	}{
+		{
+			"One Addressable",
+			[]contract.Addressable{addressable1},
+			[]contract.Addressable{addressable1},
+			false,
+		},
+		{
+			"Multiple Addressables",
+			[]contract.Addressable{addressable1, addressable2},
+
+			[]contract.Addressable{addressable1, addressable2},
+			false,
+		},
+		{
+			"No Addressables",
+			[]contract.Addressable{},
+
+			[]contract.Addressable{},
+			false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			c, err := createAndPopulateClient(tt.prePopulatedAddressables)
```

Comment:
Commented once; occurs elsewhere.

Separate `createAndPopulateClient` into `newSUT` and `populateClient`.